logbook-core/src/main/java/org/zalando/logbook/HeaderFilters.java

```diff
@@ -28,52 +31,78 @@ public static HeaderFilter defaultValue() {
 
     @API(status = MAINTAINED)
     public static HeaderFilter authorization() {
-        return replaceHeaders("Authorization"::equalsIgnoreCase, "XXX");
+        return replaceHeaders("Authorization", "XXX");
     }
 
-    public static HeaderFilter replaceHeaders(final Predicate<String> keyPredicate, final String replacement) {
-        return eachHeader((key, value) -> keyPredicate.test(key) ? replacement : value);
+    public static HeaderFilter replaceHeaders(
+            final String name,
+            final String replacement) {
+
+        return replaceHeaders(singleton(name), replacement);
     }
 
-    public static HeaderFilter replaceHeaders(final BiPredicate<String, String> predicate, final String replacement) {
-        return eachHeader((key, value) -> predicate.test(key, value) ? replacement : value);
+    public static HeaderFilter replaceHeaders(
+            final Collection<String> names,
+            final String replacement) {
+
+        return headers -> headers.apply(names, (name, previous) ->
+                previous == null ?
+                        null :
+                        nCopies(previous.size(), replacement));
     }
 
-    public static HeaderFilter eachHeader(final BinaryOperator<String> operator) {
-        return headers -> {
-            final Map<String, List<String>> result = Headers.empty();
+    public static HeaderFilter replaceHeaders(
+            final Predicate<String> keyPredicate,
+            final String replacement) {
+
+        return replaceHeaders(predicate(keyPredicate), replacement);
+    }
 
-            headers.forEach((name, values) ->
-                    result.put(name, values.stream()
-                            .map(value -> operator.apply(name, value))
-                            .collect(toList())));
+    public static HeaderFilter replaceHeaders(
+            final BiPredicate<String, String> predicate,
+            final String replacement) {
 
-            return Headers.immutableCopy(result);
-        };
+        return eachHeader((key, value) ->
+                predicate.test(key, value) ? replacement : value);
     }
 
-    public static HeaderFilter removeHeaders(final Predicate<String> keyPredicate) {
-        return removeHeaders((key, value) -> keyPredicate.test(key));
+    public static HeaderFilter eachHeader(
+            final BinaryOperator<String> operator) {
+
+        return headers -> headers.apply(headers.keySet(), (name, values) ->
+                values.stream()
```

Comment:
This could be optimized in the way that if `values.equals(result)` that we just return `values` in order not to create a new version of headers at all.